tests/pyfunc/test_inferred_data_path.py

```diff
@@ -0,0 +1,123 @@
+import pickle
+import mlflow
+import os
+from mlflow.models import Model
+import numpy as np
+import pytest
+import sklearn.neighbors
+import sklearn.datasets
+import sys
+
+sys.path.insert(0, ".")
```

Comment:
What is this line for?

Reply:
We can use https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.syspath_prepend

Reply:
We should avoid changing the global state like (`sys.path`) or environment variables. That can lead to weird and hard-to-debug errors. If you need to change it, make sure to clean up the change.

Reply:
btw, I want to run `tests/pyfunc/test_inferred_data_path.py` solely in CI, and use `tests/pyfunc` as its current working directory, :
`pushd pytest tests/pyfunc && pytest test_inferred_data_path.py && popd`

WDYT ?

Otherwise the "inferring code path" will capture all modules under `mlflow_repo`, which is annoying.
and we can set